docs/DeveloperGuide.md

```diff
@@ -1,33 +1,230 @@
+Have inquiries for us?
+Visit the FAQ section in the User Guide.
+Seeking a brief overview of all features?
+Check out the Command Summary page for a summary of all commands.
+Uncertain about a specific term's meaning?
+Refer to the Glossary page for clarification.
+
+## Quick Start
+
+To get started quickly with ActiveEdge,
+follow these simple steps:
+
+1. Ensure that you have Java 11 or above installed.
+2. Down the latest version of `ActiveEdge` from [here](http://link.to/duke). 
+3. Copy the file into the folder you want to use as the home folder for your ActiveEdge.
+4. Open a command terminal, cd into the home folder you put the jar file in, and run the following command: java -jar ActiveEdge.jar. You should see a welcome screen.
+5. Type commands beside >>> and press Enter to execute it.
+(e.g. typing ```help``` and pressing Enter will show the help page).
+## Design & implementation 🏹  
+### System Architecture
+
+![System Architecture](../images/System-Architecture.png)
+
+The Architecture Diagram above shows a high-level overview of the architectural design of ActiveEdge.
+
+Active Edge is comprised of 8 major components:
+
+```Main```: The entry point of the application.\
+```UI```: A package responsible for handling user interactions, both capturing input and displaying output.\
+```Parser```: A package that analyzes and interprets the user's input commands.\
+```Command```: A package tasked with executing specific actions based on the user's commands.\
+```Storage```: A package focused on persisting and retrieving all relevant data from the application's local storage system, ensuring data continuity and integrity.\
+```LogList```: Maintains logged data throughout the application's runtime.\
+```FoodData```: Active Edge's food database\
+```ExerciseData```: Active Edge's exercise database\
+### Generic Sequence Diagram
+The generic sequence diagram provides a visual representation of the interactions between various components within the
+ActiveEdge application. It illustrates how user commands are parsed, executed, and interact with different modules such
+as user interface, data storage, and task management.
+
+![Generic Sequence Diagram](../images/Generic-Sequence-Diagram.png)
+
+### Parser
+The Parser components can be found within the Parser package.
+It is responsible for parsing the input String of the user, and returning an appropriate XYZCommand class.
+If the input is invalid, it throws exceptions to the Error package for error handling.
+The Parser Class Diagram below shows how Execute, Parser, Error, Command classes of their respective packages work together.
+
+![Main](../images/Parser.png)
+
+### Command
+
+In the provided code, commands are represented as individual classes, each encapsulating a specific action or operation 
+within the ActiveEdge application, promoting modularity and separation of concerns.
+
+![Command](../images/Command-Class.png)
+
+
+
+
```

Comment:
UML diagrams are not present. I think the file path might be broken as I see that the images are in the other folder.
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/111181615/bc3b429e-6e5d-4ce5-9eb7-79359c9aacdc)